lib/client.js

```diff
@@ -553,6 +563,29 @@ Client.prototype.waitForSchemaAgreement = function (connection, callback) {
 };
 
 /**
+ * Waits for schema agreements and schedules schema metadata refresh.
+ * @param {Connection} connection
+ * @param event
+ * @param {Function} callback
+ * @ignore
+ * @internal
+ */
+Client.prototype.handleSchemaChange = function (connection, event, callback) {
+  var self = this;
+  this._waitForSchemaAgreement(connection, function (err) {
+    if (err) {
+      //we issue a warning but we continue with the normal flow
+      self.log('warning', 'There was an error while waiting for the schema agreement between nodes', err);
+    }
+    if (self.options.isMetadataEnabled) {
+      // schedule metadata refresh but do not wait for it to finish (no callback)
+      self.controlConnection.handleSchemaChange(event);
```

Comment:
I see... Nice catch!

The solution would be to queue it up with the rest of schema changes and process the queue immediately.